File: xoom-distributed-ping-pong/src/main/java/io/vlingo/xoom/examples/pingpong/domain/impl/PingerActor.java
package io.vlingo.xoom.examples.pingpong.domain.impl;

import io.vlingo.xoom.actors.Actor;
import io.vlingo.xoom.examples.pingpong.domain.Config;
import io.vlingo.xoom.examples.pingpong.domain.Pinger;
import io.vlingo.xoom.examples.pingpong.domain.Ponger;

public class PingerActor extends Actor implements Pinger {

  @Override
  public void ping(final Ponger ponger, final String node) {
    logger().info("... from " + node);

    try {
      Thread.sleep(1_000);
    } catch (InterruptedException e) {
      e.printStackTrace();
    }

    ponger.pong(Config.nodeName);
  }
}

File: xoom-actors-playground-kotlin/src/main/kotlin/io/vlingo/xoom/examples/pingpong/PingerActor.kt
package io.vlingo.xoom.examples.pingpong

import io.vlingo.xoom.actors.Actor
import io.vlingo.xoom.actors.testkit.TestUntil

class PingerActor: Actor, Pinger {

    var count:Int
    val pinger:Pinger
    val until:TestUntil

    constructor(until: TestUntil) {
        this.until = until
        this.count = 0
        this.pinger = selfAs(Pinger::class.java)
    }


    override fun ping(ponger: Ponger) {
        ++count
        logger().debug("ping $count")
        if (count > 10) {
            pinger.stop()
            ponger.stop()
        } else {
            ponger.pong(pinger)
        }
    }

    override fun afterStop() {
        logger().debug("Pinger: ${address()} just stopped")
        until.happened()
        super.afterStop()
    }
}